sycl/include/CL/sycl/device.hpp

```diff
@@ -20,16 +19,23 @@ namespace cl {
 namespace sycl {
 // Forward declarations
 class device_selector;
+namespace detail {
+class device_impl;
```

Comment:
```suggestion
// Forward declarations
class device_impl;
```

Reply:
@AGindinson there's the same comment two lines above. Is it necessary to repeat it twice?

Reply:
@alexbatashev Ah, the namespaces structure clarifies it for me now.